File: test-suite/src/test/java/example/micronaut/GithubController.java
package example.micronaut;

import io.micronaut.core.async.annotation.SingleResult;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.smallrye.mutiny.Uni;
import java.util.List;

@Controller("/github")
class GithubController {
    private final GithubApiClient githubApiClient;
    GithubController(GithubApiClient githubApiClient) {
        this.githubApiClient = githubApiClient;
    }

    @Get("/releases")
    @SingleResult
    Uni<List<GithubRelease>> fetchReleases() {
        return githubApiClient.fetchReleases();
    }
}

File: test-suite-kotlin/src/test/kotlin/example/micronaut/GithubController.kt
package example.micronaut

import io.micronaut.core.async.annotation.SingleResult
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.smallrye.mutiny.Uni

@Controller("/github")
class GithubController(private val githubApiClient: GithubApiClient) {

    @Get("/releases")
    @SingleResult
    fun fetchReleases(): Uni<List<GithubRelease>> {
        return githubApiClient.fetchReleases()
    }
}
